index.ts

```diff
@@ -1,35 +1,58 @@
 import HyperHTMLElement from 'hyperhtml-element';
 
+const parseAttribute = (value: string) => {
+  let parsedValue: string|number|object = value;
+  if (!isNaN(value as any)) {
+    parsedValue = +value;
+  }
+  return parsedValue;
+};
+
+const attributeName = (attr: string|BioAttribute) => typeof attr === 'string' ? attr : attr.name;
+
+export interface BioAttribute {
+  name: string;
+  converter: Function;
+}
 
 export default class BioElement<TProps extends object, TState> extends HyperHTMLElement<TState> {
 
   private _props: TProps;
 
+  // overwrite if some attributes should be auto-merged to your props
+  static bioAttributes: (string|BioAttribute)[] = [];
+
   static get observedAttributes(): string[] {
-    return [];
+    return this.bioAttributes.map(attributeName);
   };
 
-  attributeChangedCallback() {
-    this.onPropsChanged();
+  attributeChangedCallback(name: string, _: string, newValue: string): void {
+    const attribute = BioElement.bioAttributes.find(attr => attributeName(attr) === newValue);
+    this.props = {
+      ...(this.props as any),
+      [name]: typeof attribute === 'string'
+        ? parseAttribute(newValue)
```

Comment:
if I remember correctly this is not what @SheepFromHeaven wanted: if the attribute does not have a custom convert function, leave it as a string.

Reply:
From what I understood, if the developer says nothing about the attribute (i.e., if it's just a string), then we parse it. If not, we're expecting a converter function. Right?

Reply:
@SheepFromHeaven could you comment on that? You had that "an ID should stay a string even if it is just digits"-example.

Reply:
String by default, anything else custom function 👍

Reply:
Reverted 👍